tools/refseq_masher/matches.xml

```diff
@@ -0,0 +1,293 @@
+  </requirements>
+  <command detect_errors="exit_code">
+<![CDATA[
+
+#import re
+
+#if $input.type == 'fasta'
+#set $input_files = '"{}"'.format($input.fasta.name)
+  ln -s "$input.fasta" $input_files &&
+#elif $input.type == 'paired'
+#set $_forward_ext = '.fastq.gz' if $re.match(r'.*\.gz$', $input.forward.name) else '.fastq'
+#set $_forward = '"{}_1{}"'.format($re.sub(r'_[12]\..+$', '', $input.forward.name), $_forward_ext)
+#set $_reverse_ext = '.fastq.gz' if $re.match(r'.*\.gz$', $input.reverse.name) else '.fastq'
+#set $_reverse = '"{}_2{}"'.format($re.sub(r'_[12]\..+$', '', $input.reverse.name), $_reverse_ext)
+#set $input_files = '{} {}'.format($_forward, $_reverse)
+  ln -s "$input.forward" $_forward &&
+  ln -s "$input.reverse" $_reverse &&
+#elif $input.type == 'single'
+#set $input_files = '"{}"'.format($input.single.name)
+  ln -s "$input.single" $input_files &&
+#elif $input.type == 'paired_collection'
+#set $_forward = '"{}_1.{}"'.format($input.paired_collection.name, $input.paired_collection.forward.ext)
+#set $_reverse = '"{}_2.{}"'.format($input.paired_collection.name, $input.paired_collection.reverse.ext)
+#set $input_files = '{} {}'.format($_forward, $_reverse)
+  ln -s "$input.paired_collection.forward" $_forward &&
+  ln -s "$input.paired_collection.reverse" $_reverse &&
+#end if
+
+refseq_masher 
+  $adv.verbosity 
+  matches 
+  --output refseq_masher-matches.${adv.output_type}
+  --output-type $adv.output_type
+  --top-n-results $top_n_results
+#if $adv.min_kmer_threshold
+  --min-kmer-threshold $adv.min_kmer_threshold
+#end if
+  -T "\${TMPDIR:-/tmp}"
+  $input_files
+]]>
+  </command>
+  <inputs>
+    <conditional name="input">
+      <param name="type" type="select" label="Sequence input type">
+        <option value="fasta">Genome FASTA</option>
+        <option value="paired">Paired-end FASTQs</option>
+        <option value="single">Single-end FASTQ</option>
+        <option value="paired_collection">Paired-end FASTQ collection</option>
+      </param>
+      <when value="fasta">
+        <param name="fasta"
+          type="data" format="fasta"
+          optional="false"
+          label="Genome FASTA file"
+          />
+      </when>
+      <when value="paired">
+        <param name="forward"
+          type="data" format="fastq,fastqsanger,fastqillumina,fastqsolexa"
+          optional="false"
+          label="Forward FASTQ file"
+          />
+        <param name="reverse"
+          type="data" format="fastq,fastqsanger,fastqillumina,fastqsolexa"
+          optional="false"
+          label="Reverse FASTQ file"
+          help="File format must match the Forward FASTQ file"
+          />
+      </when>
+      <when value="single">
+        <param name="single"
+          type="data" format="fastq,fastqsanger,fastqillumina,fastqsolexa"
+          optional="false"
+          label="Single-end FASTQ file"
+          />
+      </when>
+      <when value="paired_collection">
```

Comment:
If I run `matches` or `contains` with a collection of paired-end files, I get the following error:

```
Fatal error: Exit code 1 ()
2018-02-14 09:20:14,049 INFO: Collected 0 FASTA inputs and 0 read sets [in /tmp/tmp7q5CL7/job_working_directory/000/22/conda-env/lib/python3.6/site-packages/refseq_masher/utils.py:185]
2018-02-14 09:20:14,049 INFO: Ran Mash dist on all input. Merging NCBI taxonomic information into results output. [in /tmp/tmp7q5CL7/job_working_directory/000/22/conda-env/lib/python3.6/site-packages/refseq_masher/cli.py:94]
Traceback (most recent call last):
  File "/tmp/tmp7q5CL7/job_working_directory/000/22/conda-env/bin/refseq_masher", line 6, in <module>
    sys.exit(refseq_masher.cli.cli())
  File "/tmp/tmp7q5CL7/job_working_directory/000/22/conda-env/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/tmp/tmp7q5CL7/job_working_directory/000/22/conda-env/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/tmp/tmp7q5CL7/job_working_directory/000/22/conda-env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/tmp/tmp7q5CL7/job_working_directory/000/22/conda-env/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmp/tmp7q5CL7/job_working_directory/000/22/conda-env/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/tmp/tmp7q5CL7/job_working_directory/000/22/conda-env/lib/python3.6/site-packages/refseq_masher/cli.py", line 95, in matches
    dfout = merge_ncbi_taxonomy_info(pd.concat(dfs))
  File "/tmp/tmp7q5CL7/job_working_directory/000/22/conda-env/lib/python3.6/site-packages/pandas/core/reshape/concat.py", line 206, in concat
    copy=copy)
  File "/tmp/tmp7q5CL7/job_working_directory/000/22/conda-env/lib/python3.6/site-packages/pandas/core/reshape/concat.py", line 239, in __init__
    raise ValueError('No objects to concatenate')
ValueError: No objects to concatenate
```

The individual files work when running through `matches` or `contains` though.

Reply:
Single end collections work 100% though.

Reply:
What's the Galaxy command for PE collections?

Reply:
Here's one of the command lines:

```
ln -s "/tmp/tmp7q5CL7/files/000/dataset_12.dat" "ASM-04_S1_L001_001_1.fastqsanger" && ln -s "/tmp/tmp7q5CL7/files/000/dataset_11.dat" "ASM-04_S1_L001_001_2.fastqsanger" && refseq_masher -vv matches --output refseq_masher-matches.tab --output-type tab --top-n-results 5 --min-kmer-threshold 8 -T "${TMPDIR:-/tmp}" "ASM-04_S1_L001_001_1.fastqsanger" "ASM-04_S1_L001_001_2.fastqsanger"
```

Reply:
Should hopefully work now in f3cd512

Reply:
Not quite.  I'm now getting:

```
Traceback (most recent call last):
  File "/tmp/tmpM4BfoB/galaxy-dev/lib/galaxy/jobs/runners/__init__.py", line 170, in prepare_job
    job_wrapper.prepare()
  File "/tmp/tmpM4BfoB/galaxy-dev/lib/galaxy/jobs/__init__.py", line 909, in prepare
    self.command_line, self.extra_filenames, self.environment_variables = tool_evaluator.build()
  File "/tmp/tmpM4BfoB/galaxy-dev/lib/galaxy/tools/evaluation.py", line 445, in build
    raise e
TypeError: expected string or buffer
```

There's no command line as Galaxy was not able to run, so I'm thinking it's maybe not able to match up/parse filenames?  The file names I have are:

`ASM-01_S1_L001_R1_001.fastq` and `ASM-01_S1_L001_R2_001.fastq`

Reply:
Your fixes look like they are all working.  Was there anything else you were going to add?

Reply:
No, I don't think so. Can you think of anything? Any nice-to-haves?

Reply:
Nope.  Going to merge it.